ssl/statem/extensions_clnt.c

```diff
@@ -453,6 +563,7 @@ EXT_RETURN tls_construct_ctos_alpn(SSL_CONNECTION *s, WPACKET *pkt,
         SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
         return EXT_RETURN_FAIL;
     }
```

Comment:
Again we have duplicated code here between the no-ech and ech code which I think we can improve.

Reply:
Same answer as above - made changes that simplify if one has a few small changes to no-ech block.